### DISCUSSION TITLE
Why cgo is needed when building golang projects using skywalking-go agent

### DISCUSSION THREAD

**Discussion by unihon** [apache/skywalking#12064] [Q&A]
When using skywalking-go agent (0.4.4), I found that it depends on gcc. I want to know what packages require gcc and whether I can disable the dependency on gcc.

build output:
```bash
# runtime/cgo
cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in $PATH
2024/04/01 14:01:07 exit status 2
```

**Comment by wu-sheng** [comment]
What is go agent 0.4.4? We don't have that version.

**Comment by unihon** [comment]
skywalking-go agent repo:
https://github.com/apache/skywalking-go/tree/v0.4.0

**Comment by wu-sheng** [comment]
We have that for a long time. https://github.com/apache/skywalking-go/pull/66

You could search the source codes and history on your own.

**Comment by Superskyyy** [comment]
It's used to extract system runtime metrics I believe.
